File: Deprecated/CommonVB/Networking/DataStream.vb

Namespace Networking
    Public Class DataStream
        Inherits IO.MemoryStream


        Private mBR As IO.BinaryReader
        Public ReadOnly Property BR() As IO.BinaryReader
            Get
                Return Me.mBR
            End Get
        End Property
        Private Sub setBR(ByVal value As IO.BinaryReader)
            Me.mBR = value
        End Sub


        Private mBW As IO.BinaryWriter
        Public ReadOnly Property BW() As IO.BinaryWriter
            Get
                Return Me.mBW
            End Get
        End Property
        Private Sub setBW(ByVal value As IO.BinaryWriter)
            Me.mBW = value
        End Sub


        Public Sub New(ByVal nBuffer As Byte(), ByVal Index As Integer, ByVal Count As Integer)
            MyBase.New(nBuffer, Index, Count)
            Me.CreateBinaryStreams()
        End Sub
        Public Sub New(ByVal nBuffer As Byte())
            Me.New(nBuffer, 0, nBuffer.Length)

        End Sub
        Public Sub New(ByVal Capacity As Integer)
            MyBase.New(Capacity)
            Me.CreateBinaryStreams()
        End Sub
        Public Sub New()
            MyBase.New()
            Me.CreateBinaryStreams()
        End Sub
        Protected Sub CreateBinaryStreams()
            Me.setBR(New IO.BinaryReader(Me))
            Me.setBW(New IO.BinaryWriter(Me))
        End Sub

        Public ReadOnly Property BytesLeft() As Long
            Get
                Return Me.Length - Me.Position
            End Get
        End Property

        Public Sub WriteToDataStream(ByVal nDataStream As DataStream, ByVal nLength As Integer)
            'Dim Buffer As Byte() = New Byte(nLength) {}
            'nDataStream.Read(Buffer, 0, nLength)
            nDataStream.Write(Me.BR.ReadBytes(nLength), 0, nLength)

        End Sub



    End Class
End Namespace

File: Common.Core/CommonOld/DataStream.cs
using System.IO;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
namespace MHGameWork.TheWizards.Common.Core
{
    public class DataStream : MemoryStream
    {


        private System.IO.BinaryReader mBR;
        public BinaryReader BR
        {
            get { return this.mBR; }
        }
        private void setBR(System.IO.BinaryReader value)
        {
            this.mBR = value;
        }


        private System.IO.BinaryWriter mBW;
        public BinaryWriter BW
        {
            get { return this.mBW; }
        }
        private void setBW(System.IO.BinaryWriter value)
        {
            this.mBW = value;
        }


        public DataStream(byte[] nBuffer, int Index, int Count)
            : base(nBuffer, Index, Count)
        {
            this.CreateBinaryStreams();
        }
        public DataStream(byte[] nBuffer)
            : this(nBuffer, 0, nBuffer.Length)
        {

        }
        public DataStream(int Capacity)
            : base(Capacity)
        {
            this.CreateBinaryStreams();
        }
        public DataStream()
            : base()
        {
            this.CreateBinaryStreams();
        }
        protected void CreateBinaryStreams()
        {
            this.setBR(new System.IO.BinaryReader(this));
            this.setBW(new System.IO.BinaryWriter(this));
        }

        public long BytesLeft
        {
            get { return this.Length - this.Position; }
        }

        public void WriteToDataStream(DataStream nDataStream, int nLength)
        {
            //Dim Buffer As Byte() = New Byte(nLength) {}
            //nDataStream.Read(Buffer, 0, nLength)
            nDataStream.Write(this.BR.ReadBytes(nLength), 0, nLength);

        }
    }
}